gobblin-utility/src/test/java/gobblin/util/AvroFlattenerTest.java

```diff
@@ -0,0 +1,303 @@
+package gobblin.util;
+
+import org.apache.avro.Schema;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class AvroFlattenerTest {
```

Comment:
Moved json inputs to resources, and added more test cases (positive and negative)